src/main/java/com/github/steveice10/mc/protocol/codec/MinecraftCodecHelper.java

```diff
@@ -91,7 +90,7 @@ public String readResourceLocation(ByteBuf buf) {
     }
 
     public void writeResourceLocation(ByteBuf buf, String location) {
-        this.writeString(buf, location);
+        this.writeString(buf, Identifier.formalize(location));
```

Comment:
I think this is unnecessary - this should be enforced at the constructor/setter level. Also, not all resource locations have to be formalized over the network.